### PR TITLE
General purpose hooks

### LINKED ISSUE #3389: Add general purpose hooks
## What is the motivation for adding / enhancing this feature?

- `afterAppInit: listener` - after modules and theme are registered
- `beforeStoreViewChange(storeView): mutator `- you can access/modify storeView object before changing
- `afterStoreViewChange(storeView): mutator`- you can access storeView object after changing storeView
- `beforePlaceOrder(order): mutator` - you can access/modify order object before sending order to backend
- `afterPlaceOrder(order, task): listener` - you can access order/task (order placement status) object after sending to backend 
- `beforeHydated(initialState):mutator` - caleld before `replaceState` and before `$mount`
- `onAfterProductThumbnailPathGenerated(path, sizex, sizey): mutator` - for integrating 3rd party services like Cloudinary,
- `onBeforeTaxesCalculated(productList)` - for integrating the 3rd party services like https://www.taxjar.com/ - it must work on kind of async/await Promises then

### PR DESCRIPTION
**Pull request by andrzejewsky**
### Related issues
closes #3389

### Short description and why it's useful
I added new hooks but most of them that were mentioned in the task are already implemented.

### Which environment this relates to
- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

### Contribution and currently important rules acceptance
- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [x] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [x] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: core/hooks.ts [hunk 1]
```diff
@@ -32,7 +44,9 @@ const coreHooks = {
   /** Hook is fired right after storeView (multistore) is changed
    * @param storeView current storeView
   */
-  afterStoreViewChange: afterStoreViewChangeHook
+  afterStoreViewChange: afterStoreViewChangeHook,
```
Comment (pkarw) [inline core/hooks.ts]: Should end with “ed” - changed :)

**Comment by kkdg** [comment]
Nice to hear that! Love to make an example off of it. Btw, when is this going to be released in version-wise? Since I am also working on versioning with `hooks` to avoid compatibility hell from the docs. @pkarw

**Comment by kkdg** [comment]
@pkarw current `hooks` recipe as of 1.10 as follows : 
https://vuepress.andflyaway.now.sh/guide/cookbook/module.html#_2-recipe

**Comment by pkarw** [comment]
@kkdg that’s perfect 👍

**Comment by pkarw** [comment]
@andrzejewsky pelase check the CI - it’s failing by some reason

**Pull request merged**